ui/intermittent-failures/GraphsContainer.jsx

```diff
@@ -92,26 +92,25 @@ export default class GraphsContainer extends React.Component {
 GraphsContainer.propTypes = {
   graphOneData: PropTypes.arrayOf(
     PropTypes.shape({
-      date: PropTypes.shape({ Date: PropTypes.string }),
-      value: PropTypes.number,
+      data: PropTypes.arrayOf(PropTypes.shape({})),
+      color: PropTypes.string,
     }),
   ),
   graphTwoData: PropTypes.arrayOf(
     PropTypes.shape({
-      date: PropTypes.shape({ Date: PropTypes.string }),
-      value: PropTypes.number,
+      data: PropTypes.arrayOf(PropTypes.shape({})),
+      color: PropTypes.string,
+    }),
+    PropTypes.shape({
+      data: PropTypes.arrayOf(PropTypes.shape({})),
+      color: PropTypes.string,
     }),
-    PropTypes.arrayOf(
-      PropTypes.shape({
-        date: PropTypes.shape({ Date: PropTypes.string }),
-        value: PropTypes.number,
-      }),
-    ),
   ),
-  children: PropTypes.element.isRequired,
+  children: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.bool]),
```

Comment:
I changed it to `children: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.bool]),` with defaut prop of `null`. I also noticed that `datePicker` is also `children: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.bool]),` in `Layout` component so i believe this is better for consistency.

I think the `PropTypes.element` is for React element as shown [here](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes) while `PropTypes.element.isReqiured` is for [requiring a single child](https://reactjs.org/docs/typechecking-with-proptypes.html#requiring-single-child). Since the `datePicker` prop seems to be used as a react element in the `GraphsContainer` component, I think using `PropTypes.element` with a default prop of `null` will work too.

@sarah-clements, I have a question. In what case would `children` be a boolean? I noticed it's a react component and I understand that `datePicker` as a class or functional component is also an object which is `PropTypes.shape({})`, but I don't know how it's possible to be a boolean. Thanks.

Reply:
Good question. My guess is that if there isn't a child prop, the prop value is falsey. Try removing the `PropType.bool` and see what kind of message you see in the console :) It's also possible something changed with the code a while back and that PropType wasn't updated.

Reply:
Removing the `PropType.bool` didn't give any message on the console. 

@sarah-clements After tracing the way it's been used, I think it should be `PropTypes.element.isReqiured` since `children` is used directly as a React element.

Also, I can't find a case where `children` wouldn't be a required `prop`. `Layout` is using `{datePicker}` directly in the `GraphsContainer` component and `datePicker` is a required `props` as it is defined both in `BugDetailsView` and `MainView` where `Layout` is used as `{<DateOptions updateState={updateState} />}`  and `{<DateRangePicker updateState={updateState} />}` resp.

I also think, `datePicker` should also be `PropTypes.element.isReqiured` in `Layout` component.

Reply:
Ok, let's go with your recommended change then.